docs/maintainer/knowledge_base.md

```diff
@@ -1655,6 +1655,37 @@ add some information on r packages which make heavy use of `noarch: generic`
     - [importlib_metadata and importlib-metadata](https://github.com/conda-forge/importlib_metadata-feedstock/blob/main/recipe/meta.yaml)
     - [typing_extensions and typing-extensions](https://github.com/conda-forge/typing_extensions-feedstock/blob/main/recipe/meta.yaml)
 
+### Common patterns
+
+#### Splitting out heavy optional dependencies
```

Comment:
Hmh, fair point, this way is a bit of mess. I found the list structure a bit annoying because it somewhat strongly enforces brevity and discourages explanations beyond a short summary, but it is just a more detailed explanation of a special case of `Distributing the same project with different sets of dependencies`. Hmm...